File: Work/LeetCode/Problems/251-Flatten2DVector/Flatten2DVector.cpp
/*************************************************************************
    > File Name: Flatten2DVector.cpp
    > Author: lwei2
    > Mail: lwei2@IT.com 
    > The Description :
 ************************************************************************/


#include <iostream>
#include <vector>

using namespace std;

class Vector2D
{
	public:
		Vector2D(vector<int>> &vec2d)
		{
			begin = vec2d.begin();
			end = vec2d.end();
			pos = 0;
		}
		int next()
		{
			hasNext();
			return (*begin)[pos++];
		}
		bool hasNext()
		{
			while(begin!=end && pos == (*begin).size())
				begin++,pos = 0;
			return begin != end;
		}
	private:
		vector<vector<int>>:;iterator begin, end;
		int pos;
};


int main(void)
{
	return 0;
}

File: Work/LeetCode/Problems/251-Flatten2DVector/Flatten2DVector.java
/*************************************************************************
    > File Name: Flatten2DVector.java
    > Author: lwei2
    > Mail: lwei2@IT.com 
    > The Description :
 ************************************************************************/

public class Vector2D
{
	List<Iterator<Iterator>> its;
	int curr = 0;
	public Vector2D(List<List<Integer>> vec2d)
	{
		this.its = new ArrayList<Iterator<Integer>>();
		for(List<Integer> l : vec2d)
		{
			if(l.size() > 0)
				this.its.add(l.iterator());
		}
	}
	public int next()
	{
		Integer res = its.get(curr).next();
		if(!its.get(curr).hasNext())
			curr++;
		return res;
	}
	public boolean hasNext()
	{
		return curr < its.size() && its.get(curr).hasNext();
	}
}
